src/lib/mdns/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -139,7 +139,66 @@ class AdvertiserMinMdns : public ServiceAdvertiser,
     /// interfaces on which the mDNS server is listening
     bool ShouldAdvertiseOn(const chip::Inet::InterfaceId id, const chip::Inet::IPAddress & addr);
 
-    FullQName GetCommisioningTextEntries(const CommissionAdvertisingParameters & params);
+    FullQName GetCommissioningTxtEntries(const CommissionAdvertisingParameters & params);
+    FullQName GetOperationalTxtEntries(const OperationalAdvertisingParameters & params);
+
+    struct CommonTxtEntryStorage
+    {
+        // CRA and CRI are both 3 chars + '=' = 4 + 1 for nullchar
+        char mrpRetryIntervalIdleBuf[kTxtRetryIntervalIdleMaxLength + 4 + 1];
+        char mrpRetryIntervalActiveBuf[kTxtRetryIntervalActiveMaxLength + 4 + 1];
+    };
+    template <class Derived>
+    CHIP_ERROR AddCommonTxtEntries(const BaseAdvertisingParams<Derived> & params, CommonTxtEntryStorage & storage,
+                                   char ** txtFields, size_t & numTxtFields)
+    {
+        Optional<uint32_t> mrpRetryIntervalIdle, mrpRetryIntervalActive;
+        params.GetMRPRetryIntervals(mrpRetryIntervalIdle, mrpRetryIntervalActive);
+        // TODO: Issue #5833 - MRP retry intervals should be updated on the poll period value change or device type change.
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+        if (chip::DeviceLayer::ConnectivityMgr().GetThreadDeviceType() ==
+            chip::DeviceLayer::ConnectivityManager::kThreadDeviceType_SleepyEndDevice)
+        {
+            uint32_t sedPollPeriod;
+            ReturnErrorOnFailure(chip::DeviceLayer::ThreadStackMgr().GetPollPeriod(sedPollPeriod));
+            // Increment default MRP retry intervals by SED poll period to be on the safe side
+            // and avoid unnecessary retransmissions.
+            mrpRetryIntervalIdle.SetValue(mrpRetryIntervalIdle.Value() + sedPollPeriod);
```

Comment:
```suggestion
            mrpRetryIntervalIdle.SetValue(mrpRetryIntervalIdle.ValueOr(0) + sedPollPeriod);
```
`Value()` may assert here.

Reply:
Good catch - thanks. I'm going to change this to only update the value if it's set in the first place so that a caller can still omit this flag even if thread is enabled.

Reply:
fixed in latest commit.